pkg/inventory/db.go

```diff
@@ -200,25 +170,119 @@ func (db *DB) GetResources(ctx context.Context) <-chan []resourceapi.Device {
+	devices := []resourceapi.Device{}
+
+	pci, err := ghw.PCI()
+	if err != nil {
+		klog.Errorf("Could not get PCI devices: %v", err)
+		return devices
 	}
-	// Set the device name. It will be normalized only if necessary.
-	device.Name = names.SetDeviceName(ifName)
-	// expose the real interface name as an attribute in case it is normalized.
-	device.Attributes["dra.net/ifName"] = resourceapi.DeviceAttribute{StringValue: &ifName}
-
-	linkType := link.Type()
-	linkAttrs := link.Attrs()
-
-	// identify the namespace holding the link as the other end of a veth pair
-	netnsid := link.Attrs().NetNsID
-	if podName := db.GetPodName(netnsid); podName != "" {
-		device.Attributes["dra.net/pod"] = resourceapi.DeviceAttribute{StringValue: &podName}
+
+	for _, pciDev := range pci.Devices {
+		if !isNetworkDevice(pciDev) {
+			continue
+		}
+		device := resourceapi.Device{
+			Name:       names.NormalizePCIAddress(pciDev.Address),
+			Attributes: make(map[resourceapi.QualifiedName]resourceapi.DeviceAttribute),
+			Capacity:   make(map[resourceapi.QualifiedName]resourceapi.DeviceCapacity),
+		}
+		device.Attributes[apis.AttrPCIAddress] = resourceapi.DeviceAttribute{StringValue: &pciDev.Address}
+		if pciDev.Vendor != nil {
+			device.Attributes[apis.AttrPCIVendor] = resourceapi.DeviceAttribute{StringValue: &pciDev.Vendor.Name}
+		}
+		if pciDev.Product != nil {
+			device.Attributes[apis.AttrPCIDevice] = resourceapi.DeviceAttribute{StringValue: &pciDev.Product.Name}
+		}
+		if pciDev.Subsystem != nil {
+			device.Attributes[apis.AttrPCISubsystem] = resourceapi.DeviceAttribute{StringValue: &pciDev.Subsystem.ID}
+		}
+
+		if pciDev.Node != nil {
+			device.Attributes[apis.AttrNUMANode] = resourceapi.DeviceAttribute{IntValue: ptr.To(int64(pciDev.Node.ID))}
+		}
+
+		pcieRootAttr, err := deviceattribute.GetPCIeRootAttributeByPCIBusID(pciDev.Address)
+		if err != nil {
+			klog.Infof("Could not get pci root attribute: %v", err)
+		} else {
+			device.Attributes[pcieRootAttr.Name] = pcieRootAttr.Value
+		}
+		devices = append(devices, device)
 	}
+	return devices
+}
+
+func (db *DB) discoverNetworkInterfaces(pciDevices []resourceapi.Device) []resourceapi.Device {
+	links, err := netlink.LinkList()
+	if err != nil {
+		klog.Errorf("Could not list network interfaces: %v", err)
+		return pciDevices
+	}
+
+	pciDeviceMap := make(map[string]*resourceapi.Device)
+	for i := range pciDevices {
+		pciDeviceMap[pciDevices[i].Name] = &pciDevices[i]
+	}
+
+	otherDevices := []resourceapi.Device{}
+
+	for _, link := range links {
+		ifName := link.Attrs().Name
+		if ignoredInterfaceNames.Has(ifName) {
+			klog.V(4).Infof("Network Interface %s is in the list of ignored interfaces, excluding it from discovery", ifName)
+			continue
+		}
+
+		// skip loopback interfaces
+		if link.Attrs().Flags&net.FlagLoopback != 0 {
+			klog.V(4).Infof("Network Interface %s is a loopback interface, excluding it from discovery", ifName)
+			continue
+		}
+
+		pciAddr := link.Attrs().ParentDev
```

Comment:
is this something reliable to obtain the pciAddres? I
maybe the`sys/net/devname` readlink  will be more reliable than depending on the netlink library and we already have `realpath()` in sysfs.go to  obtain it

Reply:
I haven't found any reason to doubt it's reliability. To me it seems like the equivalent of what we are doing manually but in this case, the kernel already maintains and provides this info directly so let's use it: https://github.com/torvalds/linux/blob/41cd3fd152634250fdd09a52a35352b3f323800d/net/core/rtnetlink.c#L2162-L2165.

Reply:
I take aback what I said above. There's a catch here which means it doesn't always work for our use case.

So ParentDev is reliable in that it gives you the immediate "parent" of the network interface. 
- This works for cases like `/sys/devices/pci0000:8c/0000:8c:00.0/0000:8d:00.0/0000:8e:02.0/0000:91:00.0/net/eth3` where the parent of eth3 is returned by linux as `0000:91:00.0` (i.e. the parent of the "net" directory). 
- But for cases like`/sys/devices/pci0000:00/0000:00:04.0/virtio1/net/eth0`, following similar logic, the OS returns the parent as `virtio1`. This DOES NOT work for us. We want `0000:00:04.0`

Even our existing logic has its own issues. I'm fixing them in https://github.com/google/dranet/pull/210 and will use that to determine the address.